### PR TITLE
Dialogue

### PR DESCRIPTION
**Pull request by duriumm**
Dialogue system first implementation

- Prototype dialogue system in place that works by making twine stories
- "Interactable" script added which can be added to any gameobject
- Dialogue Canvas for.. well the dialogue :)
- "IconPopupInRange" script added to make an icon popup when you enter a gameobjects range

### CODE REVIEW

**Comment by yousernaym** [comment]
Good work @duriumm!

Could you check line 56 in `PlayerInput.cs`? I get a null reference when I close the dialogue or if I try to open it without being near the npc.
I did some tweaks with scales and anchors to make the dialogue canvas adapt correctly to different window sizes. Also tried a dark background for the dialogue text (feel free to revert that commit if you don't like it).
In `NpcData.cs` I changed it to get components in Start() so we don't have to call GetComponent every time they are needed.

File: Pixel Game/Assets/Scripts/NpcData.cs [hunk 1]
```diff
@@ -15,16 +15,16 @@ public class NpcData : MonoBehaviour
 
 
 
-    private GameObject dialogueController;
-    private GameObject dataToPassGameObject;
+    private DialogueController dialogueController;
+    private DataToPassBetweenScenes dataToPass;
```
Comment (duriumm) [inline Pixel Game/Assets/Scripts/NpcData.cs:19]: Is there a specific reason to get the class rather than the gameobject in the scene? @yousernaym

**Comment by duriumm** [comment]
> Good work @duriumm!
> 
> Could you check line 56 in `PlayerInput.cs`? I get a null reference when I close the dialogue or if I try to open it without being near the npc.
> I did some tweaks with scales and anchors to make the dialogue canvas adapt correctly to different window sizes. Also tried a dark background for the dialogue text (feel free to revert that commit if you don't like it).
> In `NpcData.cs` I changed it to get components in Start() so we don't have to call GetComponent every time they are needed.

The null reference is because the player has not walked near the trader Npc to assign currentActiveTrader. Try walking close to the trader and then start the dialogue, no error will arise! 

This is not a biggie but will be fixed :)

File: Pixel Game/Assets/Scripts/NpcData.cs (hunk 1 above)
Comment (yousernaym) [reply Pixel Game/Assets/Scripts/NpcData.cs:19]: The GameObjects were never used anyway. It was always `dataToPassGameObject.GetComponent<DataToPassBetweenScenes>()`for example. I just moved those calls to Start()

**Comment by yousernaym** [comment]
Yep everything seems to work as it should, I just think unhandled exceptions should be avoided as a rule :)

**Comment by duriumm** [comment]
> Yep everything seems to work as it should, I just think unhandled exceptions should be avoided as a rule :)

Good point!
I'll add it to bug list and fix it soon!

**Pull request merged**